crates/flowctl/src/controlplane.rs

```diff
@@ -50,27 +66,98 @@ pub(crate) fn new_client(config: &Config) -> anyhow::Result<Client> {
     }
 }
 
-pub fn configure_new_access_token(ctx: &mut CliContext, token: String) -> anyhow::Result<()> {
+pub async fn configure_new_access_token(ctx: &mut CliContext, token: String) -> anyhow::Result<()> {
     // try to catch issues caused by missing or extra data that may have been accidentally copied
-    let email = check_access_token(&token)?;
-    ctx.config_mut().set_access_token(token);
-    println!("Configured access token for user: '{email}'");
+    let jwt = check_access_token(&token)?;
+    ctx.config_mut().set_access_token(token, jwt.sub);
+    let client = ctx.controlplane_client().await?;
+    let refresh_token = api_exec::<RefreshToken>(
+        client.rpc("create_refresh_token", r#"{"multi_use": true, "valid_for": "14d", "detail": "Created by flowctl"}"#)
+    ).await?;
+
+    if let Some(api) = &ctx.config().api {
+        persist_credentials(&api.endpoint, &api.user_id, &refresh_token).await?;
+    }
+
+    let message = if let Some(email) = jwt.email {
+        format!("Configured access token for user '{email}'")
+    } else {
+        "Configured access token".to_string()
+    };
+    println!("{}", message);
     Ok(())
 }
 
-fn check_access_token(token: &str) -> anyhow::Result<String> {
+fn check_access_token(token: &str) -> anyhow::Result<JWT> {
     let jwt = parse_jwt(token).context("invalid access_token")?;
     // Try to give users a more friendly error message if we know their credentials are expired.
     if jwt.is_expired() {
         anyhow::bail!("access token is expired, please re-authenticate and then try again");
     }
-    Ok(jwt.email)
+    Ok(jwt)
+}
+
+async fn retrieve_credentials(
+    endpoint: &url::Url,
+    user_id: &str,
+) -> anyhow::Result<RefreshToken> {
+    let keychain_entry = keychain_auth_entry(endpoint, user_id)?;
```

Comment:
I'm thinking this might be a good place to check for a `FLOW_AUTH_TOKEN` env variable. The keychain is ideal for end-users using flowctl directly on their dev machine, but some CI servers and remote dev environments may not always provide a system keychain, and an env var would be much easier to configure on something like github actions, anyway.